pkg/process/monitor/process_monitor.go

```diff
@@ -0,0 +1,285 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+// +build linux
+
+package monitor
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"runtime"
+	"sync"
+
+	"github.com/vishvananda/netlink"
+
+	"github.com/DataDog/datadog-agent/pkg/process/util"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/gopsutil/process"
+)
+
+const (
+	processMonitorMaxEvents = 2048
+)
+
+var (
+	processMonitorLock sync.RWMutex
```

Comment:
do we really need this to be thread safe?

if yes, let's use `once.Do` instead of the RwMutex. I think we can simplify by just initializing the struct once and then just accessing it

Reply:
once.Do will guarantee call once, ok. but what if GetMonitor() is called twice, the first one will initialize, the second return nil and not waiting the initialization of singleton ?